src/tests/localvocal-offline-test.cpp

```diff
@@ -441,31 +453,63 @@ int wmain(int argc, wchar_t *argv[])
 				frames_size_bytes = frames * frame_size_bytes;
 			}
 			{
-				std::lock_guard<std::mutex> lock(gf->whisper_buf_mutex);
-
-				// push back current audio data to input circlebuf
-				for (size_t c = 0; c < gf->channels; c++) {
-					circlebuf_push_back(&gf->input_buffers[c],
-							    audio[c].data() +
-								    frames_count * frame_size_bytes,
-							    frames_size_bytes);
+				bool wait = false;
+				auto max_wait =
+					start_time_time + (window_number * window_size_in_ms);
+				for (;;) {
+					{
+						std::lock_guard<std::mutex> lock(
+							gf->whisper_buf_mutex);
+						wait = gf->input_buffers->size != 0;
+					}
+					if (!wait)
+						break;
+
+					// sleep up to window size in case whisper is processing, so the buffer builds up similar to OBS
+					auto now = std::chrono::system_clock::now();
+					if (false && now > max_wait)
```

Comment:
kind of undecided if this `false` should be a parameter, or whether the max wait should be removed

for running tests deterministically between machines you currently want to always only feed a single chunk of audio into `gf->input_buffers`

honoring `max_wait` gives you something that is closer to the "plugin within obs experience", i.e. while whisper inference (etc) is running (which takes longer than the wait time on my machine), audio buffers continue to fill up, so on the next run of the whisper loop a bigger chunk of audio is fed into VAD